mlrun/model.py

```diff
@@ -1514,26 +1534,54 @@ def ui_url(self) -> str:
 
     @property
     def outputs(self):
-        """return a dict of outputs, result values and artifact uris"""
-        outputs = {}
+        """
+        Return a dictionary of outputs, including result values and artifact URIs.
+
+        This method waits for the outputs to complete and combines result values
+        and artifact URIs into a single dictionary. If there are multiple artifacts
+        for the same key, only include the artifact that does not have the "latest" tag.
+        If there is no other tag, include the "latest" tag as a fallback.
+
+        :return: Dictionary containing result values and artifact URIs.
+        """
         self._outputs_wait_for_completion()
+        outputs = {}
+
+        # Add results if available
         if self.status.results:
-            outputs = {k: v for k, v in self.status.results.items()}
+            outputs.update(self.status.results)
+
+        # Add artifacts with targets if available
+        # There are cases when `status.artifact_uri` does not exist in the run (e.g., when running locally).
         if self.status.artifacts:
-            for a in self.status.artifacts:
-                key = a["metadata"]["key"]
-                outputs[key] = get_artifact_target(a, self.metadata.project)
+            for artifact in self.status.artifacts:
+                key = artifact["metadata"]["key"]
+                tag = artifact["metadata"].get("tag")
+                if (tag != "latest") or (key not in outputs):
+                    outputs[key] = get_artifact_target(artifact, self.metadata.project)
```

Comment:
What about a case where in the same run I log the same artifact with multiple tags?
I will only have one of the tags in the output.

Like this for example:
```python
def handler(context):
    # do something first
    context.log_artifact("some-key", tag="v1")
    # do something else
    context.log_artifact("some-key", tag="v2")
    # do something else again
    context.log_artifact("some-key", tag="v3")
```
It might be an edge case but nothing stops me from doing it.

Reply:
@TomerShor Nice catch.
In this method, the `outputs["some-key"]` will include the artifact with tag `v3`. However, the problem is with the `output` function, which returns `v1` because it returns the first artifact it finds. 
This issue only occurs in the remote function, where `status.artifacts` includes 4 artifacts, but in the local function, `status.artifacts` includes only the v3 artifact.

Reply:
I updated the function and added unit and system tests for it